src/clients/detector.rs

```diff
@@ -82,6 +101,22 @@ impl ContentAnalysisRequest {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
```

Comment:
I copied these traits from `ContentAnalysisRequest`, but I am not sure this class actually needs all of them.

Reply:
The code seems to work without `Deserialize` and `Clone`. Should these be removed?
Also, `PartialEq` is only used for unit tests. Should I move it to `#[cfg_attr(test, derive(PartialEq))]` instead?